src/language-js/print/object.js

```diff
@@ -46,31 +46,27 @@ function printObject(path, options, print) {
     .sort((a, b) => locStart(a) - locStart(b))[0];
 
   const parent = path.getParentNode(0);
+
   const isFlowInterfaceLikeBody =
     isTypeAnnotation &&
     parent &&
     (parent.type === "InterfaceDeclaration" ||
       parent.type === "DeclareInterface" ||
       parent.type === "DeclareClass") &&
     path.getName() === "body";
+
   const shouldBreak =
     n.type === "TSInterfaceBody" ||
     isFlowInterfaceLikeBody ||
     (n.type === "ObjectPattern" &&
-      parent.type !== "FunctionDeclaration" &&
-      parent.type !== "FunctionExpression" &&
-      parent.type !== "ArrowFunctionExpression" &&
-      parent.type !== "ObjectMethod" &&
-      parent.type !== "ClassMethod" &&
-      parent.type !== "ClassPrivateMethod" &&
-      parent.type !== "AssignmentPattern" &&
-      parent.type !== "CatchClause" &&
```

Comment:
- [ ] TODO: ensure this functionality is tested with all of these parent types, and check if any others may be needed.